rpc/block_test.go

```diff
@@ -227,6 +231,173 @@ func TestBlockTransactionCount(t *testing.T) {
 	})
 }
 
+type MockTrie struct {
```

Comment:
Just fyi, there's no need to write mock types by hand. They can be auto-generated from interface definitions. For example, try autogenerating the StateHistoryReader interfaces mock types by executing

go generate -run "mockgen -destination=../mocks/mock_state.go -package=mocks github.com/NethermindEth/juno/core"

(or just clicking "go run generate" above the interface definition if you use VSCode).

This should regenerate mocks/mock_state.go